src/offset/local/windows_sys/mod.rs

```diff
@@ -0,0 +1,151 @@
+
+use crate::{Datelike, NaiveDateTime, Timelike};
+
+/// Chrono's internal wrapper of Window's [SYSTEMTIME][spec] structure that
+/// provides a variety of methods for interacting with Window's system calls
+/// and `SYSTEMTIME`.
+///
+/// [spec]: https://learn.microsoft.com/en-us/windows/win32/api/minwinbase/ns-minwinbase-systemtime
+pub(crate) struct WinSystemTime {
+    inner: SYSTEMTIME,
+}
+
+impl fmt::Debug for WinSystemTime {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("WinSystemTime")
+            .field("year", &self.inner.wYear)
+            .field("month", &self.inner.wMonth)
+            .field("dayOfWeek", &self.inner.wDayOfWeek)
+            .field("day", &self.inner.wDay)
+            .field("hour", &self.inner.wHour)
+            .field("minute", &self.inner.wMinute)
+            .field("second", &self.inner.wSecond)
+            .field("ms", &self.inner.wMilliseconds)
+            .finish()
+    }
+}
+
+impl WinSystemTime {
+    /// Creates a new empty `WinSystemTime`.
+    pub(crate) fn new() -> Self {
+        let st = SYSTEMTIME {
+            wYear: 0,
+            wMonth: 0,
+            wDayOfWeek: 0,
+            wDay: 0,
+            wHour: 0,
+            wMinute: 0,
+            wSecond: 0,
+            wMilliseconds: 0,
+        };
+
+        Self { inner: st }
+    }
+
+    /// Creates a `WinSystemTime` from a [`NaiveDateTime`].
+    pub(crate) fn from_naive_datetime(dt: &NaiveDateTime) -> Self {
+        let st = SYSTEMTIME {
+            // Valid values: 1601-30827
+            wYear: dt.year() as u16,
+            // Valid values:1-12
+            wMonth: dt.month() as u16,
+            // Valid values: 0-6, starting Sunday.
+            // NOTE: enum returns 1-7, starting Monday, so we are
+            // off here, but this is not currently used in local.
+            wDayOfWeek: dt.weekday() as u16,
+            // Valid values: 1-31
+            wDay: dt.day() as u16,
+            // Valid values: 0-23
+            wHour: dt.hour() as u16,
+            // Valid values: 0-59
+            wMinute: dt.minute() as u16,
+            // Valid values: 0-59
+            wSecond: dt.second() as u16,
+            // Valid values: 0-999
+            wMilliseconds: 0,
+        };
+
+        Self { inner: st }
+    }
+
+    /// Creates a `WinSystemTime` from an i64 unix seconds value.
+    pub(crate) fn from_unix_seconds(seconds: i64) -> Result<Self, Error> {
+        let t = ((seconds * HECTONANOSECS_IN_SEC) + HECTONANOSEC_TO_UNIX_EPOCH) as u64;
+        let filetime = FILETIME { dwLowDateTime: t as u32, dwHighDateTime: (t >> 32) as u32 };
+        let mut st = WinSystemTime::new();
+        unsafe { windows_sys_call!(FileTimeToSystemTime(&filetime, st.mut_inner()), 0) };
+        Ok(st)
+    }
+
+    /// Return `WinSystemTime`'s inner `SYSTEMTIME` struct.
+    pub(crate) fn inner(&self) -> SYSTEMTIME {
```

Comment:
I think we can replace with an `impl From<WinSystemTime> for NaiveDateTime`, which seems like a nicer API?